wallet/cardano-mnemonic-crypto-provider.js

```diff
@@ -12,7 +12,11 @@ const {HdNode, mnemonicToHdNode, hdNodeStringToHdNode} = require('./hd-node')
 const derivePublic = require('./helpers/derivePublic')
 const {packAddress, unpackAddress} = require('./address')
 
-const CardanoMnemonicCryptoProvider = (mnemonicOrHdNodeString, walletState) => {
+const CardanoMnemonicCryptoProvider = (
+  mnemonicOrHdNodeString,
+  walletState,
+  disableCaching = false
```

Comment:
not sure if we need this. It looks that it is there only because of tests. Would be better to create a new provider in each test case if you do not want to operate on cached keys?

Reply:
yes, I considered also this approach, but I didn't like the need to reinitialize the cryptoproviders on each test. Moreover, it is quite easy to forget about caching when writing tests if you don't explicitly manage it. For example, the test for nonhardened derivation actually was not testing the nonhardened derivation since the result was already cached from the hardened one with the same index, and I sense that there could be more similar cases in the future.